File: typescript/tests/wire/evaluationJobs.test.ts

import * as Sikaru from "../../src/api/index";
import { SikaruApi } from "../../src/Client";
import { mockServerPool } from "../mock-server/MockServerPool";

describe("EvaluationJobsClient", () => {
    
    test("list_jobs (1)", async () => {
        const server = mockServerPool.createServer();
        const client = new SikaruApi({ "maxRetries" : 0 , "apiKey" : "test" , "environment" : server.baseUrl });
        
        const rawResponseBody = { "key" : "value" };
        
        server
            .mockEndpoint()
            .get("/v1/projects/project_id/evaluation-jobs").respondWith()
            .statusCode(200).jsonBody(rawResponseBody)
                .build();

        
                        
                                const response = await client.evaluationJobs.listJobs("project_id");
                                expect(response).toEqual(rawResponseBody);
                              
                    
    });
          
    test("list_jobs (2)", async () => {
        const server = mockServerPool.createServer();
        const client = new SikaruApi({ "maxRetries" : 0 , "apiKey" : "test" , "environment" : server.baseUrl });
        
        const rawResponseBody = { };
        
        server
            .mockEndpoint()
            .get("/v1/projects/project_id/evaluation-jobs").respondWith()
            .statusCode(422).jsonBody(rawResponseBody)
                .build();

        
            await expect(async () => {
                return await client.evaluationJobs.listJobs("project_id")
            }).rejects.toThrow(Sikaru.UnprocessableEntityError);
    });
          
    test("create_job (1)", async () => {
        const server = mockServerPool.createServer();
        const client = new SikaruApi({ "maxRetries" : 0 , "apiKey" : "test" , "environment" : server.baseUrl });
        const rawRequestBody = { "evaluator" : "evaluator" , "requestId" : "requestId" , "revision" : "revision" , "rubric" : "rubric" , "targets" : [ { "target" : { "accountId" : "accountId" , "kind" : "message" } , "traceId" : "traceId" } ] };
        const rawResponseBody = { "key" : "value" };
        
        server
            .mockEndpoint()
            .post("/v1/projects/project_id/evaluation-jobs").jsonBody(rawRequestBody)
                .respondWith()
            .statusCode(200).jsonBody(rawResponseBody)
                .build();

        
                        
                                const response = await client.evaluationJobs.createJob("project_id", {
    evaluator: "evaluator",
    requestId: "requestId",
    revision: "revision",
    rubric: "rubric",
    targets: [{
            target: {
                accountId: "accountId",
                kind: "message"
            },
            traceId: "traceId"
        }]
});
                                expect(response).toEqual(rawResponseBody);
                              
                    
    });
          
    test("create_job (2)", async () => {
        const server = mockServerPool.createServer();
        const client = new SikaruApi({ "maxRetries" : 0 , "apiKey" : "test" , "environment" : server.baseUrl });
        const rawRequestBody = { "evaluator" : "x" , "requestId" : "x" , "revision" : "x" , "rubric" : "x" , "targets" : [ { "target" : { "accountId" : "x" , "kind" : "message" } , "traceId" : "x" } , { "target" : { "accountId" : "x" , "kind" : "message" } , "traceId" : "x" } ] };
        const rawResponseBody = { };
        
        server
            .mockEndpoint()
            .post("/v1/projects/project_id/evaluation-jobs").jsonBody(rawRequestBody)
                .respondWith()
            .statusCode(422).jsonBody(rawResponseBody)
                .build();

        
            await expect(async () => {
                return await client.evaluationJobs.createJob("project_id", {
    evaluator: "x",
    requestId: "x",
    revision: "x",
    rubric: "x",
    targets: [{
            target: {
                accountId: "x",
                kind: "message"
            },
            traceId: "x"
        }, {
            target: {
                accountId: "x",
                kind: "message"
            },
            traceId: "x"
        }]
})
            }).rejects.toThrow(Sikaru.UnprocessableEntityError);
    });
          
    test("get_job (1)", async () => {
        const server = mockServerPool.createServer();
        const client = new SikaruApi({ "maxRetries" : 0 , "apiKey" : "test" , "environment" : server.baseUrl });
        
        const rawResponseBody = { "key" : "value" };
        
        server
            .mockEndpoint()
            .get("/v1/projects/project_id/evaluation-jobs/job_id").respondWith()
            .statusCode(200).jsonBody(rawResponseBody)
                .build();

        
                        
                                const response = await client.evaluationJobs.getJob("project_id", "job_id");
                                expect(response).toEqual(rawResponseBody);
                              
                    
    });
          
    test("get_job (2)", async () => {
        const server = mockServerPool.createServer();
        const client = new SikaruApi({ "maxRetries" : 0 , "apiKey" : "test" , "environment" : server.baseUrl });
        
        const rawResponseBody = { };
        
        server
            .mockEndpoint()
            .get("/v1/projects/project_id/evaluation-jobs/job_id").respondWith()
            .statusCode(422).jsonBody(rawResponseBody)
                .build();

        
            await expect(async () => {
                return await client.evaluationJobs.getJob("project_id", "job_id")
            }).rejects.toThrow(Sikaru.UnprocessableEntityError);
    });
          
    test("cancel_job (1)", async () => {
        const server = mockServerPool.createServer();
        const client = new SikaruApi({ "maxRetries" : 0 , "apiKey" : "test" , "environment" : server.baseUrl });
        
        const rawResponseBody = { "key" : "value" };
        
        server
            .mockEndpoint()
            .post("/v1/projects/project_id/evaluation-jobs/job_id/cancel").respondWith()
            .statusCode(200).jsonBody(rawResponseBody)
                .build();

        
                        
                                const response = await client.evaluationJobs.cancelJob("project_id", "job_id");
                                expect(response).toEqual(rawResponseBody);
                              
                    
    });
          
    test("cancel_job (2)", async () => {
        const server = mockServerPool.createServer();
        const client = new SikaruApi({ "maxRetries" : 0 , "apiKey" : "test" , "environment" : server.baseUrl });
        
        const rawResponseBody = { };
        
        server
            .mockEndpoint()
            .post("/v1/projects/project_id/evaluation-jobs/job_id/cancel").respondWith()
            .statusCode(422).jsonBody(rawResponseBody)
                .build();

        
            await expect(async () => {
                return await client.evaluationJobs.cancelJob("project_id", "job_id")
            }).rejects.toThrow(Sikaru.UnprocessableEntityError);
    });
          
});

File: ruby/test/wire/evaluation_jobs_test.rb
# frozen_string_literal: true

require_relative "wiremock_test_case"

class EvaluationJobsWireTest < WireMockTestCase
  def setup
    super

    @client = Sikaru::Client.new(
      token: "<token>",
      base_url: WIREMOCK_BASE_URL
    )
  end

  def test_evaluation_jobs_list_jobs_with_wiremock
    test_id = "evaluation_jobs.list_jobs.0"

    @client.evaluation_jobs.list_jobs(
      project_id: "project_id",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "evaluation_jobs.list_jobs.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "GET",
      url_path: "/v1/projects/project_id/evaluation-jobs",
      query_params: nil,
      expected: 1
    )
  end

  def test_evaluation_jobs_create_job_with_wiremock
    test_id = "evaluation_jobs.create_job.0"

    @client.evaluation_jobs.create_job(
      project_id: "project_id",
      evaluator: "evaluator",
      request_id: "requestId",
      revision: "revision",
      rubric: "rubric",
      targets: [{
        target: {
          account_id: "accountId",
          kind: "message"
        },
        trace_id: "traceId"
      }],
      request_options: {
        additional_headers: {
          "X-Test-Id" => "evaluation_jobs.create_job.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "POST",
      url_path: "/v1/projects/project_id/evaluation-jobs",
      query_params: nil,
      expected: 1
    )
  end

  def test_evaluation_jobs_get_job_with_wiremock
    test_id = "evaluation_jobs.get_job.0"

    @client.evaluation_jobs.get_job(
      project_id: "project_id",
      job_id: "job_id",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "evaluation_jobs.get_job.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "GET",
      url_path: "/v1/projects/project_id/evaluation-jobs/job_id",
      query_params: nil,
      expected: 1
    )
  end

  def test_evaluation_jobs_cancel_job_with_wiremock
    test_id = "evaluation_jobs.cancel_job.0"

    @client.evaluation_jobs.cancel_job(
      project_id: "project_id",
      job_id: "job_id",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "evaluation_jobs.cancel_job.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "POST",
      url_path: "/v1/projects/project_id/evaluation-jobs/job_id/cancel",
      query_params: nil,
      expected: 1
    )
  end
end
